File: problems/rice/Prob04.java
package problems.rice;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;
import problems.Input;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/**
 *
 * @author rootie
 */
public class Prob04 {

    public static void main(String[] args) throws FileNotFoundException {
        // get file obj
        Scanner scanner = new Scanner(new File(Input.JPATH + "Prob04.in.txt"));
        int testCaseAmt = Integer.parseInt(scanner.nextLine());
        for (int tCa = 0; tCa < testCaseAmt; tCa++) {
            int subTestCaseAmt = Integer.parseInt(scanner.nextLine());
            for (int i = 0; i < subTestCaseAmt; i++) {
                String line = scanner.nextLine();
                if (line.split(" ")[1].equals("C")) {
                    double f = (Double.parseDouble(line.split(" ")[0]) / (5.0 / 9.0)) + 32.0;
                    System.out.println(line + " = " + String.valueOf(Math.round(f * 10.0) / 10.0) + " F");
                } else {
                    double c = (5.0 / 9.0) * (Double.parseDouble(line.split(" ")[0]) - 32.0);
                    System.out.println(line + " = " + String.valueOf(Math.round(c * 10.0) / 10.0) + " C");
                }
            }
        }
        scanner.close();
    }
}

File: problems/rice_py/prob04.py
import path
import math

def c(f):
	return float((5.0 / 9.0) * (f - 32.0))

def f(c):
	return float((((9.0 / 5.0) * c) + 32.0))

path = path.j_input + "Prob04.in.txt"
file = open(path, "r")
amt = int(file.readline())

for i in range(0, amt):
	sa = int(file.readline())
	for i2 in range(0, sa):
		line = file.readline()
		line = line[0:len(line) - 1]
		b = line.split(' ')
		if (b[1] == "F"):
			new = c(float(b[0]))
			print(line + " = " + str(math.ceil(new * 10) / 10) + " C")
		else:
			new = f(float(b[0]))
			print(line + " = " + str(math.ceil(new * 10) / 10) + " F")
			
		
